xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-api/src/main/java/org/xwiki/search/solr/internal/api/SolrIndexer.java

```diff
@@ -20,6 +20,7 @@
 package org.xwiki.search.solr.internal.api;
 
 import org.xwiki.component.annotation.Role;
+import org.xwiki.link.ReadyIndicator;
```

Comment:
Feels strange to use a `org.xwiki.link.*` for something which is way more generic than links (we have quite a few use cases which would benefit from this new indicator instead of the current queue size based hack which is used).

`org.xwiki.link.ReadyIndicator` should probably be a wrapper on link API side calling a `org.xwiki.link.ReadyIndicator` returned by `SolrIndexer`.

Reply:
> `org.xwiki.link.ReadyIndicator` should probably be a wrapper on link API side calling a `org.xwiki.link.ReadyIndicator` returned by `SolrIndexer`.

Is it intentional that you used the same class name twice? I agree regarding the package name being strange, what you suggest makes sense. Do you think that the API itself is okay? I wonder if it really makes sense to have a `Future<Boolean>` as now there are two error cases: returning `FALSE` and throwing an exception, which is not good.

Reply:
> Is it intentional that you used the same class name twice?
>  calling a `org.xwiki.link.ReadyIndicator` returned by SolrIndexer.

Sorry, was too quick, was of course `org.xwiki.solr.`. Now I'm wondering if we cannot find some more generic version that both store APIs could expose, like a `org.xwiki.store.ReadyIndicator` (in `xwiki-platform-sore-api` for example).

> `Future<Boolean>`

The Boolean values look a bit weird to me, actually. I would expect any problem with a Future to go through an exception and the value to be something which represent a successful result (but we don't have much in this case to return I guess).

Reply:
> The Boolean values look a bit weird to me, actually. I would expect any problem with a Future to go through an exception and the value to be something which represent a successful result (but we don't have much in this case to return I guess).

I found there is `Future<Void>` for this purpose, I've switched to it.

Reply:
I've moved the `ReadyIndicator` to `xwiki-platform-store-api` now, how do you like that?

Reply:
Much better :)